### PR TITLE
add `{}`,`<>` config.toml brackets for auto_close behavior

### PR DESCRIPTION
**Pull request by spaceunifyfifty**
context: https://github.com/zed-industries/zed/issues/10720#issuecomment-2063529599

### CODE REVIEW

File: languages/kotlin/config.toml [hunk 1]
```diff
@@ -3,5 +3,7 @@ grammar = "kotlin"
 path_suffixes = ["kt"]
 line_comments = ["// "]
 brackets = [
-		{ start = "(", end = ")", close = true, newline = false}
+		{ start = "(", end = ")", close = true, newline = false },
+		{ start = "{", end = "}", close = true, newline = false },
+	    { start = "<", end = ">", close = true, newline = false },
```
Comment (osiewicz) [inline languages/kotlin/config.toml:8]: I'm not an expert, but won't autoclosing `<` interfere with comparison operators?

Comment (osiewicz) [reply languages/kotlin/config.toml]: Yeah, this config is not contextual; we have pretty rudimentary ways to exclude rules (see [rust config](https://github.com/zed-industries/zed/blob/main/crates/languages/src/rust/config.toml#L10) where we exclude autobrackets from strings and comment contexts), but nothing like "when in generic context".

**Comment by evrsen** [comment]
Thank you!

File: languages/kotlin/config.toml (hunk 1 above)
Comment (JosephTLyons) [reply languages/kotlin/config.toml]: Some of our languages have <> configured and [we've gotten some complaints about it](https://github.com/zed-industries/zed/issues/10512).

**Comment by osiewicz** [comment]
Hey, I'm sorry for a late reply; yes, you're right in that we should add both `brackets.scm` and `indents.scm`; the former is responsible for highlighting matching brackets and the latter is what you're after - it's responsible for indenting new lines.
I think we can copy what we have for `{}` verbatim from Javascript's indents.scm and brackets.scm.

**Comment by osiewicz** [comment]
Well, that PR has not been merged yet, so I think it's fine to add brackets.scm in that PR and place (via rebase) a commit with that prior to the one that bumps version to 0.0.3

**Pull request merged**